server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -363,16 +363,59 @@ private boolean poolOverlapsRange(Pool p, Date startDate, Date endDate) {
         return page;
     }
 
+    /**
+     * Deletes the given entitlement.
+     *
+     * @param entity
+     *  The entitlement entity to delete
+     */
     @Transactional
     public void delete(Entitlement entity) {
         Entitlement toDelete = find(entity.getId());
-        log.debug("Deleting entitlement: {}", toDelete);
-        log.debug("certs.size = {}", toDelete.getCertificates().size());
 
-        for (EntitlementCertificate cert : toDelete.getCertificates()) {
-            currentSession().delete(cert);
+        if (toDelete != null) {
+            this.deleteImpl(toDelete);
+
+            // Maintain runtime consistency.
+            entity.getCertificates().clear();
+            entity.getConsumer().getEntitlements().remove(entity);
+            entity.getPool().getEntitlements().remove(entity);
+        }
+    }
+
+    /**
+     * Deletes the given collection of entitlements.
+     * <p/></p>
+     * Note: Unlike the standard delete method, this method does not perform a lookup on an entity
+     * before deleting it.
+     *
+     * @param entitlements
+     *  The collection of entitlement entities to delete
+     */
+    public void batchDelete(Collection<Entitlement> entitlements) {
+        for (Entitlement entitlement : entitlements) {
+            this.deleteImpl(entitlement);
+
+            // Maintain runtime consistency.
+            entitlement.getCertificates().clear();
+            entitlement.getConsumer().getEntitlements().remove(entitlement);
+            entitlement.getPool().getEntitlements().remove(entitlement);
```

Comment:
@Ceiu similar fix was made for master  in #1262 , which this undoes.
tbh I should have asked @nguyenfilip  to add a comment here so a future dev does not delete it inadvertently. could you add it in this PR please? if not, I can create a separate PR .